### PR TITLE
expression: fix incorrect error handle in builtInTimeFormat

### PR DESCRIPTION
**Pull request by francis0407**
### What problem does this PR solve? 
Block https://github.com/pingcap/tidb/pull/13807.

### Check List 

Tests 

 - No code

Side effects

 - Possible performance regression

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -374,7 +374,7 @@ func (b *builtinTimeFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.C
 	}
 	defer b.bufAllocator.put(buf)
 	if err := b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
-		return err
+		return vecEvalStringByRows(b, input, result)
```
Comment (wjhuang2016) [inline expression/builtin_time_vec.go]: This logic is quite strange, please add some comment for this change.

Comment (francis0407) [reply expression/builtin_time_vec.go]: addressed

**Pull request merged**